atari-DQN/replay_memory.py

```diff
@@ -113,12 +129,19 @@ def encode_states(self, idxs: np.ndarray) -> (torch.tensor, torch.tensor):
                 else seq_idxs
 
             # Fill states
+            # NOTE reshaping stacks multiple channels (if present) AND frames
+            #      into the 2nd dimension of the returned state tensor
             if len(cur_seq_idxs) > 0:
-                cur_states[i, -len(cur_seq_idxs):] = \
-                    self._obs_buffer[cur_seq_idxs]
+                cur_states[i, -len(cur_seq_idxs):] = torch.reshape(
+                    self._obs_buffer[cur_seq_idxs],
+                    (-1, self._obs_shape[-1], self._obs_shape[-1])
+                )
```

Comment:
For each non-zero-padded observations, need to stack them along with their channels. 
Make a separate variable to account for how many observations to stack (and how many channels for each observation):
```python
if len(cur_seq_idxs) > 0:
    # number of (non zero-padded) frames and channels to stack
    non_pad_len = len(cur_seq_idxs) * self._obs_shape[0]
    cur_states[i, -non_pad_len:] = torch.reshape(
        self._obs_buffer[cur_seq_idxs],
```

Reply:
Note: this is fixed in the RL-767 SMiRL repository